recipes/input4mips-validation/meta.yaml

```diff
@@ -0,0 +1,96 @@
+{% set name = "input4mips-validation" %}
+{% set version = "0.5.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/input4mips_validation-{{ version }}.tar.gz
+  sha256: e1e31e0279621023bd0e5468eb02ba89661a9a0b685cd4b46f5dfa54634b121d
+
+build:
+  entry_points:
+    - input4mips-validation = input4mips_validation.cli:app
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
+    - python >=3.9,<4.0
+    - poetry-core
+    - pip
+  run:
+    - python >=3.9.0,<4.0.0
+    - loguru >=0.7
+    - cf_xarray >=0.8
+    - xarray >=2022.0
+    - pint-xarray >=0.3
+    - numpy >=1.23,<2.0
+    - cftime >=1.6
+    - typing-extensions >=4.0
+    - pooch >=1.0
+    - netcdf4 >=1.0
+    - attrs >=22.0
+    - typer >=0.9.0,<0.10.0
+    - validators >=0.28.3,<0.29.0
+    - cattrs >=23.2.3,<24.0.0
+  run_constrained:
+    - matplotlib-base >=3.0
+    - notebook >=6.0
+
+test:
+  imports:
+    - input4mips_validation
+  commands:
+    - pip check
+    - input4mips-validation --help
+  requires:
+    - pip
+
+outputs:
+  - name: input4mips-validation
+
+  # A locked version, which will help avoid installation headaches
+  # for those who just want the CLI.
```

Comment:
Smart! First time I see this type of output, not sure how I feel about it 🤔 How is this helping with "installation headaches". Fearing a breaking dependency update or something?

Reply:
So the idea is that you have a locked version of the package, which is insensitive to dependency updates. For people who want to use the package's CLI as an application, this locked version provides a stable interface that is insensitive to dependency updates. We'll have lots of users who will just run away if they get installation issues, so I want to offer them a path that is as likely to work as possible (and avoid breaking due to dependency updates). For people who want to use the package as a library, they can still use the non-locked version with the full flexibility it provides.

As a concrete example, the locked version would mean that, even if numpy updates to 3.0, if a user installs input4mips-validation-locked, the package would still work (the latest numpy update to 2.0 has caused chaos throughout our stack, and having these locked versions would have helped a lot). 

As another reference for this kind of behaviour, there is a plugin for this in the Python world (https://pdm-build-locked.readthedocs.io/en/stable/), I am basically implementing this behaviour by hand here because the stability is key for this project.

Reply:
> They thing is that they'll probably need to install it in a new environment to maximize their chances of not running into conflicts with other projects due to the hard pinning

Correct.

> At that point, isn't something like PyInstaller or nuitka more suited?

Perhaps. For me, there's a few issues with these solutions and I would really appreciate merging with the locked version as is. Even if it isn't the perfect solution, it is a solution. A bit more reasoning:

1. Our community knows and uses conda. I had never heard of PyInstaller and nuitka and I would appreciate the ability to avoid asking our community to become familiar with another tool.
  - You are completely correct that this solution is basically using conda a bit like "pipx". I know that's not the intended use case, but I think it does work as a fine solution and I don't see any great harm (but please correct me if I'm wrong)?  (Arguably, a pre-built docker image would also be a great solution, as I said, I'm trying to avoid Docker because our community just isn't ready for it yet.)
3. Distributing with PyInstaller/nuitka looks like a case of "build somewhere, send files to others to use". Building for multiple platforms requires doing multiple builds on different platforms I assume. I don't have my head around this for those tools (I assume it is not so hard, but spending more hours working out the CI etc. for that is a step I would love to be able to not do right now). On the other hand, I do know how to make a multi-platform compatible tool with conda.
4. It's not obvious to me that PyInstaller and nuitka work with conda specific dependencies. Their docs all focus on PyPI dependencies. So, while they might work for conda, if they don't work out of the box, I'm going to have quite a bit of learning to do. As I said, I would love to avoid that.

That's obviously a fairly long answer. As I said, I completely agree that there are probably other ways to do this. I am happy to accept that those ways might be better. I hope you can see that this way should also achieve the desired goals and hence we can merge this as is.

Reply:
No one else opposed so let's see. In the future if we see that it's not as useful as intended it can be removed once in the feedstock.

If you want an example of PyInstaller / nuitka, for the sake of curiosity, here are two examples:

- PyInstaller: https://github.com/conda-forge/conda-standalone-feedstock
- nuitka: https://github.com/conda/conda-standalone/pull/80

Reply:
Very cool, thank you!